### ISSUE TITLE
Document optional parameters

### ISSUE THREAD

**Issue by SebastienGllmt** [NomicFoundation/hardhat-ignition#672]
## Problem

I want to have parameters that are optional

For example, something like this:

```ts
const additionalOwner = m.getParameter('additionalOwner');
if (additionalOwner != null) {
    m.call(token, "addOwner", [additionalOwner]);
}
```

## Current State

I don't think I can leave it undefined either, because `getParameter` needs to be of type `SolidityParameterType`, but neither `null` or `undefined` are valid `SolidityParameterType` values (see below)

https://github.com/NomicFoundation/hardhat-ignition/blob/bed5bd6f74f9f6749358244a6f02ea145a80f6c5/packages/core/src/types/module.ts#L330-L340

## Possible solutions

Other than adding default/undefined as an option for these types, another option could be a separate function `m.hasParameter`

**Comment by kanej** [comment]
I don't think we would support conditional parameters in Ignition modules. In general we try and avoid conditional logic within a module. I realise a parameter result is different from a call onchain, but I would worry that reconciliation in particular would become more complicated.
